sdk/core/azure-core/src/http/request.cpp

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <map>
+#include <string>
+
+#include <http/http.hpp>
+
+using namespace azure::core::http;
+
+http_method::HttpMethod Request::getMethod() { return this->_method; }
+std::string const& Request::getUrl() { return this->_url; }
+std::string const& Request::getBody() { return this->_body; }
+std::map<std::string, std::string> const& Request::getHeaders() { return this->_headers; }
+
+void Request::addHeader(std::string const& name, std::string const& value)
+{
+  this->_headers.insert(std::pair<std::string, std::string>(name, value));
```

Comment:
Using std::make_pair() - would eliminate the need to write `<std::string, std::string>`

Reply:
You would need to `#include <utility>` for make_pair (http://www.cplusplus.com/reference/utility/make_pair/?kw=make_pair)